vhdl_lang/src/config.rs

```diff
@@ -88,7 +90,58 @@ impl LibraryConfig {
             }
         }
         // Remove duplicate file names from the result
-        result.into_iter().unique().collect()
+        result = result.into_iter().unique().collect();
+        // Gather exclded files
+        let mut exclude: Vec<PathBuf> = Vec::new();
+        for pattern in self.exclude_patterns.iter() {
+            let stripped_pattern = if cfg!(windows) {
+                pattern.strip_prefix("\\\\?\\").unwrap_or(pattern.as_str())
+            } else {
+                pattern.as_str()
+            };
+
+            if is_literal(stripped_pattern) {
+                let file_path = PathBuf::from(pattern);
+
+                if file_path.exists() {
+                    exclude.push(file_path);
+                } else {
+                    messages.push(Message::warning(format! {"File {pattern} does not exist"}));
+                }
+            } else {
+                match glob::glob(stripped_pattern) {
+                    Ok(paths) => {
+                        let mut empty_pattern = true;
+
+                        for file_path_or_error in paths {
+                            empty_pattern = false;
+                            match file_path_or_error {
+                                Ok(file_path) => {
+                                    exclude.push(file_path);
+                                }
+                                Err(err) => {
+                                    messages.push(Message::error(err.to_string()));
+                                }
+                            }
+                        }
+
+                        if empty_pattern {
+                            messages.push(Message::warning(format!(
+                                "Pattern '{stripped_pattern}' did not match any file"
+                            )));
+                        }
+                    }
+                    Err(err) => {
+                        messages.push(Message::error(format!("Invalid pattern '{pattern}' {err}")));
+                    }
+                }
+            }
+        }
```

Comment:
There is now a lot of code duplication as this code is basically the same as the code that generates the file patterns in the first place. I suggest to factor out the main logic into a separate function  and generate the two file lists from that function.

Reply:
I've extracted the logic into a function. Would you typically add tests for this new function too, though it's result is tested in the tests for its callers?

Reply:
I generally wouldn't write a separate test-case for this if it's already tested extensively through other means, so I think this is fine.